index.js

```diff
@@ -26,19 +29,30 @@ module.exports = function (src, dest, opts) {
 
 	opts = objectAssign({overwrite: true}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
-
-		read.on('error', function (err) {
-			reject(new CpFileError('cannot read from `' + src + '`: ' + err.message, err));
-		});
+	var progress = {
+		stat: null,
+		emitter: new EventEmitter()
+	};
```

Comment:
Hence we have `cpFile.on(…)`, `stat` and `emitter` aren't intended _only_ for progress stuff anymore.
We should drop `progress.…` by:
```js
var stat = null;
var eventEmitter = new EventEmitter();
```